src/components/Button.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import styled from 'styled-components';
+
+interface Props {
+  type: 'button' | 'cancel';
```

Comment:
I think this is a good approach. But I would make it primary | button.

Reply:
This has been updated to be `primary | button` option types